cells/src/commonMain/kotlin/com/wire/kalium/cells/domain/usecase/RefreshCellAssetStateUseCase.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.cells.domain.usecase
+
+import com.wire.kalium.cells.domain.CellAttachmentsRepository
+import com.wire.kalium.cells.domain.CellsRepository
+import com.wire.kalium.cells.domain.model.CellNode
+import com.wire.kalium.cells.domain.model.NodePreview
+import com.wire.kalium.common.error.CoreFailure
+import com.wire.kalium.common.error.NetworkFailure
+import com.wire.kalium.common.error.NetworkFailure.ServerMiscommunication
+import com.wire.kalium.common.error.StorageFailure
+import com.wire.kalium.common.functional.Either
+import com.wire.kalium.common.functional.flatMap
+import com.wire.kalium.common.functional.left
+import com.wire.kalium.common.functional.map
+import com.wire.kalium.common.functional.onFailure
+import com.wire.kalium.common.functional.onSuccess
+import com.wire.kalium.common.functional.retry
+import com.wire.kalium.common.functional.right
+import com.wire.kalium.logic.data.asset.AssetTransferStatus
+import com.wire.kalium.logic.data.message.CellAssetContent
+import com.wire.kalium.logic.data.message.localPath
+import com.wire.kalium.network.exceptions.KaliumException
+import io.ktor.http.HttpStatusCode
+import okio.FileSystem
+import okio.Path.Companion.toPath
+import okio.SYSTEM
+
+/**
+ * Use case to refresh the cell asset state and preview.
+ * - Check if the asset is still available.
+ * - Remove local data if the asset is not found.
+ * - Remove local data if the asset is updated (based on contentHash).
+ * - Fetch preview URL with retries.
+ */
+public interface RefreshCellAssetStateUseCase {
+    public suspend operator fun invoke(assetId: String, fetchPreview: Boolean): Either<CoreFailure, Unit>
+}
+
+internal class RefreshCellAssetStateUseCaseImpl internal constructor(
```

Comment:
Would be great to cover that with unit tests, not sure if you already plan to do that in the future 😄